openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -235,47 +211,85 @@ def get_cohort(user, course_key, assign=True):
         if not assign:
             return None
 
-    choices = course.auto_cohort_groups
-    if len(choices) > 0:
-        # Randomly choose one of the auto_cohort_groups, creating it if needed.
-        group_name = local_random().choice(choices)
+    course_cohorts = get_course_cohorts(course, assignment_type=CourseCohort.RANDOM)
+    if course_cohorts:
+        course_cohort = local_random().choice(course_cohorts)
     else:
-        # Use the "default cohort".
-        group_name = DEFAULT_COHORT_NAME
+        course_cohort, __ = CourseUserGroup.objects.get_or_create(
```

Comment:
another variant of cohort creation which i think should be consolidated into a single method (see comment further below)